cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -185,8 +190,16 @@ public void schedule(Id.Program programId, SchedulableProgramType programType, I
         throw Throwables.propagate(e);
       }
       for (Schedule schedule : schedules) {
-        String scheduleName = schedule.getName();
-        String cronEntry = schedule.getCronEntry();
+        Preconditions.checkArgument(schedule instanceof TimeSchedule || !(schedule instanceof StreamSizeSchedule));
+        TimeSchedule timeSchedule;
+        if (!(schedule instanceof TimeSchedule)) {
+          timeSchedule = (TimeSchedule) Schedules.createTimeSchedule(schedule.getName(), schedule.getDescription(),
```

Comment:
Because ``Schedules.create...`` return ``Schedule`` objects, not ``TimeSchedule`` or other.